thoth/user_api/api_v1.py

```diff
@@ -104,6 +108,34 @@
 _PROVENANCE_CHECK_PROTECTED_FIELDS = frozenset({"kebechet_metadata"})
 
 
+CALLBACK_SECRET_NAME_TEMPLATE = "callback-{document_id}"
+
+
+def _add_item_to_callback_secret_entry(
+    document_id, callbackurl, auth_header: Optional[str] = None, client_data: Optional[dict] = None
+):
+    value = json.dumps({"callbackurl": callbackurl, "Authorization": auth_header, "client_data": client_data})
+    entry_name = random.choices(string.ascii_letters, k=16)
+    body = [{"op": "add", "path": f"/data/{entry_name}", "value": value}]
+    k8.client.CoreV1Api.patch_namespaced_secret(
+        name=CALLBACK_SECRET_NAME_TEMPLATE.format(document_id=document_id),
+        namespace=Configuration.THOTH_MIDDLETIER_NAMESPACE,
```

Comment:
As this would be called from adviser function, it would more convienent to have the secret placed in the backend namespace, unless having them in middletier help with some further action.

Reply:
++